### PR TITLE
Add Overall Difficulty (OD) query in beatmapset search

### LINKED ISSUE #7783: OD filter doesn't work
Screenshot_2021-06-23-20-23-50-217_com android chrome
Screenshot_2021-06-23-20-23-58-972_com android chrome
Screenshot_2021-06-23-20-24-05-620_com android chrome
Web map filters in general works,but not this specific one, unless I'm missing the specific keyword?

### PR DESCRIPTION
**Pull request by gagahpangeran**
closes #7783

### CODE REVIEW

**Review by peppy — DISMISSED** [review]
Looks sane

File: app/Libraries/Search/BeatmapsetSearch.php [hunk 1]
```diff
@@ -264,6 +264,7 @@ private function addSimpleFilters(BoolQuery $query, BoolQuery $nested): void
             'difficultyRating' => ['field' => 'beatmaps.difficultyrating', 'type' => 'range'],
             'drain' => ['field' => 'beatmaps.diff_drain', 'type' => 'range'],
             'hitLength' => ['field' => 'beatmaps.hit_length', 'type' => 'range'],
+            'od' => ['field' => 'beatmaps.diff_overall', 'type' => 'range'],
```
Comment (nanaya) [inline app/Libraries/Search/BeatmapsetSearch.php:267]: the key used here should match the one used in transformer (`accuracy`)

File: app/Libraries/Search/BeatmapsetSearchParams.php [hunk 2]
```diff
@@ -58,6 +58,9 @@ class BeatmapsetSearchParams extends SearchParams
     /** @var int|null */
     public $mode = null;
 
+    /** @var array|null */
+    public $od;
```
Comment (nanaya) [inline app/Libraries/Search/BeatmapsetSearchParams.php:62]: use native type declaration for new codes

File: app/Libraries/Search/BeatmapsetSearchRequestParams.php [hunk 3]
```diff
@@ -227,6 +227,7 @@ private function parseQuery(): void
             'dr' => 'drain',
             'keys' => 'keys',
             'length' => 'hitLength',
+            'od' => 'od',
```
Comment (nanaya) [inline app/Libraries/Search/BeatmapsetSearchRequestParams.php:230]: similarly here should map to match the attribute in transformer (although looking again, the normalization should've been done in query parser instead)

**Review by nanaya — CHANGES_REQUESTED** [review]

**Review by nanaya — APPROVED** [review]

**Pull request merged**